File: check_fcm_tokens.dart
import 'package:firebase_core/firebase_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'firebase_options.dart';

void main() async {
  // Khởi tạo Firebase
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );

  // Query Firestore collection 'users' với shopId
  final querySnapshot = await FirebaseFirestore.instance
      .collection('users')
      .where('shopId', isEqualTo: 'honC8KnKhOUG19wcYOFDTGVdKWP2')
      .get();

  // In ra fcmToken của mỗi user
  for (var doc in querySnapshot.docs) {
    final data = doc.data();
    final fcmToken = data['fcmToken'];
    final userId = doc.id;
    print('User ID: $userId, FCM Token: $fcmToken');
  }
}

File: functions/check_fcm_tokens.js
const admin = require('firebase-admin');

// Khởi tạo Firebase Admin SDK
admin.initializeApp();

async function checkFCMTokens() {
  const db = admin.firestore();

  // Query collection 'users' với shopId
  const querySnapshot = await db.collection('users')
    .where('shopId', '==', 'honC8KnKhOUG19wcYOFDTGVdKWP2')
    .get();

  // In ra fcmToken của mỗi user
  querySnapshot.forEach((doc) => {
    const data = doc.data();
    const fcmToken = data.fcmToken;
    const userId = doc.id;
    console.log(`User ID: ${userId}, FCM Token: ${fcmToken}`);
  });
}

checkFCMTokens().catch(console.error);